### PR TITLE
Practical 6

### PR DESCRIPTION
**Pull request by harshpatel-javascript**
Screenshot from 2022-03-22 11-59-46

### CODE REVIEW

File: Practical-6/src/components/UserList.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        <div className="user-status">Active</div>
+      ) : (
+        <div className="user-status">
+          <select name="status" id="status" onChange={dropDownMenuHandler}>
+            <option value="inactive">Inactive</option>
+            <option value="active">Active</option>
+          </select>
+        </div>
+      )}
+      {data.access === "Owner" ? (
```
Comment (desaianiruddh) [inline Practical-6/src/components/UserList.js]: you should set conditional rendering 
data.access === "Owner" 
? first element
: second element

File: Practical-6/src/css/UserContainer.css [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.user-container {
+  padding: 1rem 2rem;
```
Comment (desaianiruddh) [inline Practical-6/src/css/UserContainer.css:2]: use px instead of rem if all other place you are using px

File: Practical-6/src/index.css [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  margin: 0;
+  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
+    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
+    sans-serif;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+  font-size: 14px;
+}
+
+code {
```
Comment (desaianiruddh) [inline Practical-6/src/index.css:11]: remove code css-class which is came with CRA if there's no need

File: Practical-6/src/redux/reducer.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  PASS_DATA_TO_CARD,
+  TOGGLE_STATUS,
+  MOUSE_LEAVE,
+  DELETE_USER,
+} from "./types";
+import { userData } from "../components/userData";
+
+const initialState = {
+  cardData: false,
+  definedData: userData,
```
Comment (desaianiruddh) [inline Practical-6/src/redux/reducer.js]: Data is not defined. we are getting dynamic data from API server. you should change name of property

File: Practical-6/src/redux/reducer.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case PASS_DATA_TO_CARD:
+      return {
+        cardData: action.payload,
+        definedData: userData,
```
Comment (desaianiruddh) [inline Practical-6/src/redux/reducer.js]: use copy of previous state and modify that.
definedData is constant no need to re-write
return {
        ...state,
        cardData: action.payload
};

File: Practical-6/src/components/UserList.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        onMouseEnter={onMouseEnterHandler}
+        onMouseLeave={onMouseLeaveHandler}
+      >
+        <img src={data.avatar} alt="user" className="user-img" />
+        <div className="user-detail">
+          <p className="user-name">{data.name}</p>
+          <p className="user-email">{data.email}</p>
+        </div>
+      </div>
+      {data.status === "Active" ? (
```
Comment (desaianiruddh) [inline Practical-6/src/components/UserList.js]: you should set conditional rendering like below,
data.access === "Owner"
? first element
: second element

File: Practical-6/src/components/UserList.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import { Lock, Trash2 } from "react-feather";
+import { useDispatch } from "react-redux";
+import {
```
Comment (desaianiruddh) [inline Practical-6/src/components/UserList.js:4]: put blank line between internal and external lib.

File: Practical-6/src/components/UserList.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+function UserList({ data }) {
+  const dispatch = useDispatch();
+  const onMouseEnterHandler = () => {
+    dispatch(passDataToCard({ data }));
+  };
+  const onMouseLeaveHandler = () => {
+    dispatch(onMouseLeave());
+  };
+  const clickHandler = () => {
```
Comment (Rutvi25) [inline Practical-6/src/components/UserList.js:21]: naming convention could be more relevant like deleteHandler instead of clickHandler

**Review by Rutvi25 — CHANGES_REQUESTED** [review]

**Review by desaianiruddh — APPROVED** [review]

**Review by Rutvi25 — APPROVED** [review]

**Pull request closed without merging**